packages/apollo/src/utils/client.ts

```diff
@@ -0,0 +1,324 @@
+import { ApolloClient, from } from '@apollo/client'
+import { createUploadLink } from 'apollo-upload-client'
+import isEqual from 'lodash/isEqual'
+import { useMemo, useState } from 'react'
+
+import { isSSR } from '@open-condo/miniapp-utils/helpers/environment'
+import { useEffectOnce } from '@open-condo/miniapp-utils/hooks/useEffectOnce'
+import { usePrevious } from '@open-condo/miniapp-utils/hooks/usePrevious'
+
+
+import { addCacheToClient, createApolloCache } from './cache'
+import { createPersistor } from './cachePersistor'
+
+import { APOLLO_STATE_PROP_NAME } from '../constants'
+
+import type { InitCacheConfig, InitCacheOptions, InvalidationCacheConfig } from './cache'
+import type { CachePersistor } from './cachePersistor'
+import type { ApolloLink, NormalizedCacheObject } from '@apollo/client'
+
+type ApolloClientOptions = {
+    /** Apollo server url, can be static string or obtained from function on the fly */
+    uri: (() => string) | string
+    /** Headers to include in each request, can be helpful in SSR environments */
+    headers?: Record<string, string>
+    /** Client's cache config */
+    cacheConfig?: InvalidationCacheConfig
+    /** Middleware to process / enrich requests */
+    middlewares?: Array<ApolloLink>
```

Comment:
Will there be some set of ready-made middlewares provided for developing our applications (or not only ours)? For example, an error handler or something like that?

Reply:
Yep, they'll be probably placed in miniapp-utils package